addons/mail/static/src/discuss/core/common/discuss_core_common_service.js

```diff
@@ -40,39 +40,12 @@ export class DiscussCoreCommon {
                 });
             }
         });
-        this.busService.subscribe("discuss.channel/delete", (payload, { id: notifId }) => {
+        this.busService.subscribe("discuss.channel/delete", (payload, metadata) => {
             const thread = this.store.Thread.insert({
                 id: payload.id,
                 model: "discuss.channel",
             });
-            const filteredStarredMessages = [];
-            let starredCounter = 0;
-            for (const msg of this.store.discuss.starred.messages) {
-                if (!msg.thread?.eq(thread)) {
-                    filteredStarredMessages.push(msg);
-                } else {
-                    starredCounter++;
-                }
-            }
-            this.store.discuss.starred.messages = filteredStarredMessages;
-            if (notifId > this.store.discuss.starred.counter_bus_id) {
-                this.store.discuss.starred.counter -= starredCounter;
-            }
-            this.store.discuss.inbox.messages = this.store.discuss.inbox.messages.filter(
-                (msg) => !msg.thread?.eq(thread)
-            );
-            if (notifId > this.store.discuss.inbox.counter_bus_id) {
-                this.store.discuss.inbox.counter -= thread.message_needaction_counter;
-            }
-            this.store.discuss.history.messages = this.store.discuss.history.messages.filter(
-                (msg) => !msg.thread?.eq(thread)
-            );
-            thread.closeChatWindow?.();
-            if (thread.eq(this.store.discuss.thread)) {
-                this.store.discuss.inbox.setAsDiscussThread();
-            }
-            thread.messages.splice(0, thread.messages.length);
-            thread.delete();
+            this._handleNotificationChannelDelete(thread, metadata);
```

Comment:
This base method doesn't do anything (except potentially creating an empty thread)? Do we actually need to override it instead of moving it?

Reply:
It actually calls `_handleNotificationChannelDelete`.